cli/lib/constants/options.js

```diff
@@ -0,0 +1,120 @@
+  [camelNames.browser]: 'runs Cypress in the browser with the given name. if a filesystem path is supplied, Cypress will attempt to use the browser at that path.',
+  [camelNames.cacheClear]: 'delete all cached binaries',
+  [camelNames.cachePrune]: 'deletes all cached binaries except for the version currently in use',
+  [camelNames.cacheList]: 'list cached binary versions',
+  [camelNames.cachePath]: 'print the path to the binary cache',
+  [camelNames.cacheSize]: 'Used with the list command to show the sizes of the cached folders',
+  [camelNames.ciBuildId]: 'the unique identifier for a run on your CI provider. typically a "BUILD_ID" env var. this value is automatically detected for most CI providers',
+  [camelNames.component]: 'runs component tests',
+  [camelNames.config]: 'sets configuration values. separate multiple values with a comma. overrides any value in cypress.config.{js,ts,mjs,cjs}.',
+  [camelNames.configFile]: 'path to script file where configuration values are set. defaults to "cypress.config.{js,ts,mjs,cjs}".',
+  [camelNames.detached]: 'runs Cypress application in detached mode',
+  [camelNames.dev]: 'runs cypress in development and bypasses binary check',
+  [camelNames.e2e]: 'runs end to end tests',
+  [camelNames.env]: 'sets environment variables. separate multiple values with a comma. overrides any value in cypress.config.{js,ts,mjs,cjs} or cypress.env.json',
+  [camelNames.exit]: 'keep the browser open after tests finish',
+  [camelNames.forceInstall]: 'force install the Cypress binary',
+  [camelNames.global]: 'force Cypress into global mode as if its globally installed',
+  [camelNames.group]: 'a named group for recorded runs in Cypress Cloud',
+  [camelNames.headed]: 'displays the browser instead of running headlessly',
+  [camelNames.headless]: 'hide the browser instead of running headed (default for cypress run)',
+  [camelNames.key]: 'your secret Record Key. you can omit this if you set a CYPRESS_RECORD_KEY environment variable.',
+  [camelNames.parallel]: 'enables concurrent runs and automatic load balancing of specs across multiple machines or processes',
+  [camelNames.port]: 'runs Cypress on a specific port. overrides any value in cypress.config.{js,ts,mjs,cjs}.',
+  [camelNames.project]: 'path to the project',
+  [camelNames.quiet]: 'run quietly, using only the configured reporter',
+  [camelNames.record]: 'records the run. sends test results, screenshots and videos to Cypress Cloud.',
+  [camelNames.reporter]: 'runs a specific mocha reporter. pass a path to use a custom reporter. defaults to "spec"',
+  [camelNames.reporterOptions]: 'options for the mocha reporter. defaults to "null"',
+  [camelNames.spec]: 'runs specific spec file(s). defaults to "all"',
+  [camelNames.tag]: 'named tag(s) for recorded runs in Cypress Cloud',
+  [camelNames.version]: 'prints Cypress version',
+}
```

Comment:
@KostiantynPopovych Thank you for taking on this change! At a high-level glance, it appears this PR is tackling two things:
1. the bug where we report the wrong flag when a missing flag value was provided
2. refactoring to allow these up these flags to live in one place

This is might make this review a little slower, since a lot changes. My first pass looking at this is restructuring this data would make it a bit clearer to understand the rest of the changes. Something like the follow would be easier to reason with:

```js
const commandOptions = {
  'options_as_camelCase': {
     description:
     flag:
     shortHand:    
   }
}
```
so
```js
const commandOptions = {
  exit: {
      description: '...',
      flag: '--no-exit',
      // don't define shore flag since there isn't one.
  },
  reporterOptions: {
     description: '...',
     flag: '--reporter-options',
     shortFlag: '-o', 
   },
}
```